pkgs/development/python-modules/unify/default.nix

```
@@ -0,0 +1,22 @@
+{ lib, buildPythonPackage, fetchFromGitHub, untokenize }:
+
+buildPythonPackage rec {
+  pname = "unify";
+  version = "0.5";
+
+  # PyPi release is missing tests (see https://github.com/myint/unify/pull/18)
+  src = fetchFromGitHub {
+    owner = "myint";
+    repo = "unify";
+    rev = "v${version}";
+    sha256 = "1l6xxygaigacsxf0g5f7w5gpqha1ava6mcns81kqqy6vw91pyrbi";
+  };
+
+  propagatedBuildInputs = [ untokenize ];
```

```suggestion
  propagatedBuildInputs = [ untokenize ];
  
  checkPhase = ''
    ${python.interpreter} -m unittest discover
  '';
```

Explicitly call the test suite and don't rely on `python setup.py test`, which is deprecated.

Requires you to import `python` in the function signature.

---

Thanks, I was just looking this up, as I was struggling to understand how `setuptools` was able to find the testsuite without it being specified. I guess `setuptoolsCheckHoock` should then be updated soon as well.